agent/api/task/task.go

```diff
@@ -459,66 +460,39 @@ func (task *Task) PostUnmarshalTask(cfg *config.Config,
-func (task *Task) initNetworkMode() {
-	if len(task.ENIs) > 0 {
+// ACS is streaming down this value with task payload. In case of docker bridge mode task, this value might be left empty
+// as it's the default task network mode.
+func (task *Task) initNetworkMode(acsTaskNetworkMode *string) {
+	switch aws.StringValue(acsTaskNetworkMode) {
+	case AWSVPCNetworkMode:
 		task.NetworkMode = AWSVPCNetworkMode
-		return
-	}
-	for _, c := range task.Containers {
-		// ACS sets task def network mode to each and every customer container docker host config, so the presence
-		// of one container with a given network mode is enough to infer the task network mode
-		switch c.GetNetworkModeFromHostConfig() {
-		case BridgeNetworkMode:
-			task.NetworkMode = BridgeNetworkMode
-			return
-		case HostNetworkMode:
-			task.NetworkMode = HostNetworkMode
-			return
-		case "":
-			// the absence of a network mode means the default docker network mode is used, which, for ECS it's bridge.
-			// for future proofing, let's keep inspecting the rest of the containers for this particular case.
-			continue
-		}
+	case HostNetworkMode:
+		task.NetworkMode = HostNetworkMode
+	case BridgeNetworkMode, "":
+		task.NetworkMode = BridgeNetworkMode
+	case networkModeNone:
+		task.NetworkMode = networkModeNone
+	default:
+		logger.Warn("Unmapped task network mode", logger.Fields{
+			field.TaskID:      task.GetID(),
+			field.NetworkMode: aws.StringValue(acsTaskNetworkMode),
+		})
 	}
-	// If we reached this part, it means none of the containers has network mode set, which means bridge mode (default)
-	task.NetworkMode = BridgeNetworkMode
+	logger.Info("Task network mode initialized", logger.Fields{
+		field.TaskID:      task.GetID(),
+		field.NetworkMode: task.NetworkMode,
+	})
 }
 
 func (task *Task) initServiceConnectResources() error {
-	// TODO [SC]: ServiceConnectConfig will come from ACS. Adding this here for dev/testing purposes only Remove when
-	// ACS model is integrated
-	if task.ServiceConnectConfig == nil {
-		task.ServiceConnectConfig = &serviceconnect.Config{
-			ContainerName: "service-connect",
-		}
-	}
 	if task.IsServiceConnectEnabled() {
-		// TODO [SC]: initDummyServiceConnectConfig is for dev testing only, remove it when final SC model from ACS is in place
-		task.initDummyServiceConnectConfig()
 		if err := task.initServiceConnectEphemeralPorts(); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// TODO [SC]: This is for dev testing only, remove it when final SC model from ACS is in place
-func (task *Task) initDummyServiceConnectConfig() {
-	scContainer := task.GetServiceConnectContainer()
-	if _, ok := scContainer.Environment["SC_CONFIG"]; !ok {
-		// no SC_CONFIG :(
-		return
-	}
-	if err := json.Unmarshal([]byte(scContainer.Environment["SC_CONFIG"]), task.ServiceConnectConfig); err != nil {
-		logger.Error("Error parsing SC_CONFIG", logger.Fields{
-			field.Error: err,
-		})
-		return
-	}
-}
-
```

Comment:
Can we save these deletions until SC is available in some prod form to facilitate testing still?

Reply:
Sure